workshops/test/base.py

```diff
@@ -299,7 +299,8 @@ def _setUpEvents(self):
         self.num_uninvoiced_events = 0
         self.num_upcoming = 0
         for e in Event.objects.all():
-            if e.invoice_status == 'not-invoiced' and e.start < today:
+            e.is_past_event = e.start < today and (e.end is None or e.end < today)
```

Comment:
There was a difference in logic between the statistics we generate here and `Event.objects.uninvoiced_events().count()`. The method `uninvoiced_events()` also filters out events which have a `end` date in the future or nulled. I replicated here and `self.num_uninvoiced_events` now returns the correct number of events which haven't been invoiced.